topdown/strings.go

```diff
@@ -411,28 +412,50 @@ func builtinTrimSpace(a ast.Value) (ast.Value, error) {
 
 func builtinSprintf(a, b ast.Value) (ast.Value, error) {
 	s, err := builtins.StringOperand(a, 1)
+
 	if err != nil {
 		return nil, err
 	}
 
 	astArr, ok := b.(*ast.Array)
+
 	if !ok {
 		return nil, builtins.NewOperandTypeErr(2, b, "array")
 	}
 
 	args := make([]interface{}, astArr.Len())
 
+	var formatStringArr []string
+	if len(args) > 0 {
+		// Split up the format string
+		regex, err := regexp.Compile(`%(((\.[0-9]+)|([0-9]+)|([0-9]+\.[0-9]+))([a-zA-Z])|([a-zA-Z]))`)
```

Comment:
[nit] let's make that a package-level (non-exported) variable, like
```go
var formatRE = regexp.MustCompile(`...`)
```
to avoid that it's recompiled in every invocation -- it never changes after all.

Reply:
Fixed.